Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/mail/MailUtils.java

```diff
@@ -47,146 +47,171 @@
-    }
+	/**
+	 * Return a mail session using the provided properties, username and
+	 * password.
+	 *
+	 * @param mailServerProps
+	 *            properties for the mail server
+	 * @param user
+	 *            username credential
+	 * @param pass
+	 *            password credential
+	 * @return mail session.
+	 */
+	public static Session getMailSession(Properties mailServerProps, String user, String pass) {
+		return Session.getInstance(mailServerProps, getMailAuthenticator(user, pass));
+	}
 
-    /**
-     * Send a mime message.
-     * @param recipients of the mime message.
-     * @param session used to send the mime message.
-     * @param message to be sent.
-     * @throws MessagingException if there is an error.
-     */
-    public static void sendMessage(Address[] recipients, Session session, MimeMessage message)
-            throws MessagingException {
+	/**
+	 * Send a mime message.
+	 *
+	 * @param recipients
+	 *            of the mime message.
+	 * @param session
+	 *            used to send the mime message.
+	 * @param message
+	 *            to be sent.
+	 * @throws MessagingException
+	 *             if there is an error.
+	 */
+	public static void sendMessage(Address[] recipients, Session session, MimeMessage message)
+			throws MessagingException {
 
-        Transport transport = null;
-        try {
-            transport = session.getTransport("smtp");
-            String host = session.getProperty("mail.smtp.host");
-            String port = session.getProperty("mail.smtp.port");
-            String user = session.getProperty("connect.mail.user");
-            String pass = session.getProperty("connect.mail.pass");
-            if (!StringUtils.isBlank(port)) {
-                port = StringUtils.trim(port);
-            }
+		Transport transport = null;
+		try {
+			transport = session.getTransport("smtp");
+			String host = session.getProperty("mail.smtp.host");
+			String port = session.getProperty("mail.smtp.port");
+			String user = session.getProperty("connect.mail.user");
+			String pass = session.getProperty("connect.mail.pass");
+			if (!StringUtils.isBlank(port)) {
+				port = StringUtils.trim(port);
+			}
 
-            transport.connect(host, Integer.parseInt(port), user, pass);
-            transport.sendMessage(message, recipients);
-            logHeaders(message);
-        } catch (AssertionError e) {
-            LOG.error("Assertion Error while sending.", e);
-            throw e;
-        } finally {
-            if (transport != null) {
-                transport.close();
-            }
-        }
-    }
+			transport.connect(host, Integer.parseInt(port), user, pass);
+			transport.sendMessage(message, recipients);
+			logHeaders(message);
+		} catch (AssertionError e) {
+			LOG.error("Assertion Error while sending.", e);
+			throw e;
+		} finally {
+			if (transport != null) {
+				try {
+					transport.close();
+				} catch (Exception e) {
```

Comment:
Is 'java.lang.Exception' the actual type being thrown, or can this catch be made more granular?